lib/src/contextual-menu/ContextualMenu.test.js

```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import { render, fireEvent } from "@testing-library/react";
+import DxcContextualMenu from "./ContextualMenu.tsx";
+
+const items = [{ label: "Item 1" }, { label: "Item 2" }, { label: "Item 3" }, { label: "Item 4" }];
+
+const sections = [
+  {
+    title: "Team repositories",
+    items: [{ label: "Approved locations" }, { label: "Approved locations" }, { label: "Approved locations" }],
+  },
+  {
+    items: [{ label: "Approved locations" }, { label: "Approved locations" }, { label: "Approved locations" }],
+  },
+];
+
+describe("Context menu component tests", () => {
+  test("Context menu renders with correct aria attributes", () => {
+    const { getAllByRole, getByRole } = render(<DxcContextualMenu items={items} defaultSelectedItemIndex={0} />);
+    const renderedItems = getAllByRole("menuitem");
+    expect(renderedItems.length).toBe(4);
+    expect(renderedItems[0].getAttribute("aria-selected")).toBeTruthy();
+    expect(getByRole("menu")).toBeTruthy();
+  });
+  test("Context menu (with sections) renders with correct aria attributes", () => {
+    const { getAllByRole, getByRole } = render(<DxcContextualMenu items={sections} defaultSelectedItemIndex={4} />);
+    const renderedItems = getAllByRole("menuitem");
+    expect(renderedItems.length).toBe(6);
+    expect(renderedItems[4].getAttribute("aria-selected")).toBeTruthy();
+    expect(getAllByRole("group").length).toBe(2);
+  });
+  test("onSelect event from each item is called correctly", () => {
+    const test = [
+      {
+        label: "Tested item",
+        onSelect: jest.fn(),
+      },
+    ];
+    const { getByRole } = render(<DxcContextualMenu items={test} />);
+    const item = getByRole("menuitem");
+    fireEvent.click(item);
+    expect(test[0].onSelect).toHaveBeenCalled();
+  });
```

Comment:
The following tests should be added:
- Keyboard navigation (Arrow keys, focus and selection).
- Edge cases (Empty menu, menu with only 1 item, default selected item out of bounds...).

Reply:
I follow your structure to answer:

- There's no focus management or functionality from our side. This is the usual browser tab sequence, so there is no need to test in this case.
- Since we document these scenarios and type the props to avoid those edge cases, I consider that there is no need to over-program. If the component breaks the user should know that it is his fault (this is our approach since we added Typescript).